integration-tests/oidc-wiremock/src/main/java/io/quarkus/it/keycloak/SlackCodeFlowResource.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.it.keycloak;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.SecurityContext;
+
+import io.quarkus.oidc.UserInfo;
+import io.quarkus.security.Authenticated;
+
+@Path("/slack")
+public class SlackCodeFlowResource {
+
+    public record SlackResponseDto(String userPrincipalName, String userInfoEmail) {
+    }
+
+    @Inject
+    UserInfo userInfo;
```

Comment:
@michalvavrik FYI, this is an optional piece, with Slack, you should be able to get it from the injected `@IdToken JsonWebToken`, avoiding an extra remote UserInfo call, we only need it for OAuth2 providers which do not return ID token

Reply:
@sberyozkin yes, I'll do that. My test actually tested user info endpoint with JSON structure as returned by Slack. Your proposal lefts it untested, I suppose it is standard and we don't need that.

Reply:
Unless you mean that we should implement something special for Slack that checks ID token for user info which I don't want to do.

Reply:
Let's try to conclude it today, I'll try to move this to special module and migrate to generic OIDC wiremock resource. We can iterate later.

Reply:
I have tried and and I don't see it. That new generated ID token that is not same as original Slack ID token doesn't have it or I missed something. I'll keep user info and remove it on demand.

Reply:
Hi @michalvavrik, so if IdToken returns all the info expected by scopes such as `profile` and `email`, then a UserInfo call is redundant, it is just an extra remote call.